### PR TITLE
WebVid.

### PR DESCRIPTION
**Pull request by knighton**
## Description of changes:

WebVid processing and stub streaming dataset.

## Merge Checklist:
_Put an `x` without space in the boxes that apply. If you are unsure about any checklist, please don't hesitate to ask. We are here to help! This is simply a reminder of what we are going to look for before merging your pull request._

### General
- [x] I have read the [contributor guidelines](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md)
- [ ] This is a documentation change or typo fix. If so, skip the rest of this checklist.
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the MosaicML team.
- [x] I have updated any necessary documentation, including [README](https://github.com/mosaicml/streaming/blob/main/README.md) and [API docs](https://github.com/mosaicml/streaming/tree/main/docs) (if appropriate).

### Tests
- [x] I ran `pre-commit` on my change. (check out the `pre-commit` section of [prerequisites](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#prerequisites))
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate).
- [ ] I ran the tests locally to make sure it pass. (check out [testing](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#running-tests))
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes.

### CODE REVIEW

File: streaming/multimodal/convert/webvid.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Download videos, creating an MDS dataset.
+
+Instructions:
+
+1. Navigate to the download section of https://m-bain.github.io/webvid-dataset/, where you will
+   find 2.5M and 10M dataset splits:
```
Comment (dakinggg) [inline streaming/multimodal/convert/webvid.py:9]: is it 2.5M or 2M?

File: streaming/multimodal/webvid.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 MosaicML Streaming authors
```
Comment (dakinggg) [inline streaming/multimodal/webvid.py:1]: I'm not sure this file is worth merging, given it has `# Processing goes here.`? Maybe at least raise a not implemented error or something?

**Review by dakinggg — APPROVED** [review]

File: streaming/multimodal/convert/webvid.py (hunk 1 above)
Comment (knighton) [reply streaming/multimodal/convert/webvid.py:9]: hmm. googling around, i see both terms being used. it is 2.5m samples

File: streaming/multimodal/webvid.py (hunk 2 above)
Comment (knighton) [reply streaming/multimodal/webvid.py:1]: this class included as a demonstration of where to modify in order to do your own on-the-fly sample processing

**Pull request merged**